modules/frontend/pipeline/trace_query_filter.go

```diff
@@ -20,47 +22,35 @@ func NewTraceQueryFilterWare(next http.RoundTripper) http.RoundTripper {
 	}
 }
 
-func NewTraceQueryFilterWareWithDenyList(denyList []string) Middleware {
-	filter := make([]*regexp.Regexp, len(denyList)+1)
+func NewTraceQueryFilterWareWithDenyList(denyList []string, parseFunc func(r *http.Request) string) Middleware {
+	var filter []*regexp.Regexp
 	for i := range denyList {
 		exp, err := regexp.Compile(denyList[i])
 		if err == nil {
-			filter[i] = exp
+			filter = append(filter, exp)
 		}
 	}
 
 	return MiddlewareFunc(func(next http.RoundTripper) http.RoundTripper {
 		return traceQueryFilterWare{
 			next:    next,
 			filters: filter,
+			parseFn: parseFunc,
 		}
 	})
 }
 
 func (c traceQueryFilterWare) RoundTrip(req *http.Request) (*http.Response, error) {
-	resp, err := c.next.RoundTrip(req)
-	if err != nil {
-		return resp, err
-	}
-	//Better way to do this?
-	if len(c.filters) == 0 {
-		return resp, nil
-	}
-	//need wait group
-	u, err := api.ParseSearchRequest(req)
-	if err != nil {
-		return resp, err
+	if c.filters == nil || len(c.filters) == 0 {
```

Comment:
in go `len(<nil slice>) == 0`

```suggestion
	if len(c.filters) == 0 {
```